Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -268,31 +249,539 @@ def embProdEmbOfIsAlgebraic [Algebra E K] [IsScalarTower F E K] (halg : Algebra.
+      (hy.map (f := algebraMap F F⟮x⟯) |>.of_dvd (minpoly.dvd_map_of_isScalarTower F L y)))
+  let M := L⟮y⟯
+  letI : Algebra L M := Subalgebra.algebra M.toSubalgebra
+  letI : Module L M := Algebra.toModule
+  letI : SMul L M := Algebra.toSMul
+  haveI : IsScalarTower F L M := IntermediateField.isScalarTower M
+  haveI : FiniteDimensional F L := adjoin.finiteDimensional hx.isIntegral
+  haveI : FiniteDimensional L M := adjoin.finiteDimensional halg'.isIntegral
+  rw [finSepDegree_mul_finSepDegree_of_isAlgebraic F L M (Algebra.IsAlgebraic.of_finite L M),
+    FiniteDimensional.finrank_mul_finrank F L M] at heq
+  change finSepDegree F (restrictScalars F M) = finrank F (restrictScalars F M) at heq
+  rwa [adjoin_adjoin_left F {x} {y}, Set.union_comm, Set.union_singleton] at heq
+
+end IntermediateField
+
+/-- If `E / F` and `K / E` are both separable extensions, then `K / F` is also separable. -/
+theorem IsSeparable.trans [Algebra E K] [IsScalarTower F E K]
+    [IsSeparable F E] [IsSeparable E K] : IsSeparable F K := (isSeparable_def F K).2 fun x ↦ by
+  let f := minpoly E x
+  let E' : IntermediateField F E := adjoin F f.frange
+  haveI : FiniteDimensional F E' := finiteDimensional_adjoin fun x _ ↦ IsSeparable.isIntegral F x
+  have h : f ∈ lifts (algebraMap E' E) := (lifts_iff_coeff_lifts f).2 fun n ↦ Set.mem_range.2 <| by
+    by_cases h : f.coeff n = 0
+    · simp only [h, _root_.map_eq_zero, exists_eq]
+    have : F⟮f.coeff n⟯ ≤ E' := adjoin.mono _ _ _ <|
+      Set.singleton_subset_iff.2 <| f.coeff_mem_frange n h
+    exact ⟨⟨f.coeff n, this <| mem_adjoin_simple_self F (f.coeff n)⟩, rfl⟩
+  obtain ⟨g, h⟩ := f.mem_lifts.1 h
+  have hx : x ∈ restrictScalars F E'⟮x⟯ := mem_adjoin_simple_self _ x
+  have hzero : aeval x g = 0 := by
+    simpa only [← h, aeval_map_algebraMap] using minpoly.aeval E x
+  have halg : IsIntegral E' x := (IsSeparable.isAlgebraic F E).trans
+    (IsSeparable.isAlgebraic E K) x |>.isIntegral.tower_top
+  have hsep : f.Separable := IsSeparable.separable E x
+  rw [← h, separable_map] at hsep
+  replace hsep := hsep.of_dvd <| minpoly.dvd _ _ hzero
+  haveI : IsSeparable F E' := isSeparable_tower_bot_of_isSeparable F E' E
+  haveI := (isSeparable_adjoin_simple_iff_separable _ _).2 hsep
+  haveI := adjoin.finiteDimensional halg
+  letI : Algebra E' E'⟮x⟯ := Subalgebra.algebra E'⟮x⟯.toSubalgebra
+  letI : Module E' E'⟮x⟯ := Algebra.toModule
+  letI : SMul E' E'⟮x⟯ := Algebra.toSMul
+  haveI : IsScalarTower F E' E'⟮x⟯ := IntermediateField.isScalarTower E'⟮x⟯
+  haveI : FiniteDimensional F E'⟮x⟯ := FiniteDimensional.trans F E' E'⟮x⟯
+  have := finSepDegree_mul_finSepDegree_of_isAlgebraic F E' E'⟮x⟯ (IsSeparable.isAlgebraic _ _)
+  rw [finSepDegree_eq_finrank_of_isSeparable F E',
+    finSepDegree_eq_finrank_of_isSeparable E' E'⟮x⟯,
+    FiniteDimensional.finrank_mul_finrank F E' E'⟮x⟯,
+    eq_comm, finSepDegree_eq_finrank_iff F E'⟮x⟯] at this
+  change IsSeparable F (restrictScalars F E'⟮x⟯) at this
```

Comment:
Do you need this line (`this` is not used after).

Reply:
This is an instance used in the following `separable_of_mem_isSeparable`.